texar/modules/berts/bert_base.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Base class for Bert Modules.
+"""
+
+from texar.module_base import ModuleBase
+from texar.modules.berts import bert_utils
+
+__all__ = [
+    "BertBase",
+]
+
+
+class BertBase(ModuleBase):
+    r"""Base class for all Bert classes to inherit.
+
+    Args:
+        pretrained_model_name (optional): a str with the name
+            of a pre-trained model to load selected in the list of:
+            `bert-base-uncased`, `bert-large-uncased`, `bert-base-cased`,
+            `bert-large-cased`, `bert-base-multilingual-uncased`,
+            `bert-base-multilingual-cased`, `bert-base-chinese`.
+            If `None`, will use the model name in :attr:`hparams`.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    def __init__(self,
+                 pretrained_model_name=None,
+                 cache_dir=None,
+                 hparams=None):
+        ModuleBase.__init__(self, hparams)
```

Comment:
Prefer using `super().__init__(hparams)` instead of explicitly writing the base class name. This could lead to problems in case we change the class hierarchies.